app/controllers/assignments_controller.rb

```diff
@@ -59,19 +58,35 @@ def destroy
   end
 
   def find_volunteer
+    set_client
     @q = policy_scope(Volunteer).seeking_clients.ransack(params[:q])
     @q.sorts = ['created_at desc'] if @q.sorts.empty?
     @seeking_clients = @q.result.paginate(page: params[:page])
   end
 
   def find_client
+    set_volunteer
     @q = policy_scope(Client).need_accompanying.ransack(params[:q])
     @q.sorts = ['created_at desc'] if @q.sorts.empty?
     @need_accompanying = @q.result.paginate(page: params[:page])
   end
 
+  def last_submitted_hours_and_feedbacks
+    @last_submitted_hours = @assignment.hours_since_last_submitted
+    @last_submitted_feedbacks = @assignment.feedbacks_since_last_submitted
+  end
+
   private
 
+  def update_redirect
```

Comment:
Actually, to be honest, that whole thing with the redirect is kind of difficult. 


Why not create an own action, because it wouldn't need anything unsafe with referers, plus you'd not need to pass get variables with that form button at all. In fact you could just make it a link with the method put, like the delete links we had. So no form needed.

```ruby
# AssignmentsController
def mark_viewed
   # Do your mark viewed stuff and have full control, not accepting some get parmeters maybe not in your control
   redirect_to back_to_your_feedback_list, notice: what ever special notice ou want to use
end

# config/routes.rb

resources :volunteers do
    # ...
    resources :group_offers do
      resources :feedbacks do
        put :mark_viewed, on: :member
      end
      resources :trial_feedbacks do
        put :mark_viewed, on: :member
      end
    end
    resources :assignments do
      resources :feedbacks do do
        put :mark_viewed, on: :member
      end
      resources :trial_feedbacks do
        put :mark_viewed, on: :member
      end
    end
  end
```

or even make it shorter with a routes concern:

```ruby
# routes.rb
concern :mark_view do
   put :mark_viewed, on: :member
end

resources :volunteers do
    # ...
    resources :group_offers do
      resources :feedbacks, concerns: :mark_view
      resources :trial_feedbacks, concerns: :mark_view
    end
    resources :assignments do
      resources :feedbacks, concerns: :mark_view
      resources :trial_feedbacks, concerns: :mark_view
    end
  end
```